website/docs/concepts/scopes.mdx

```diff
@@ -0,0 +1,331 @@
+        // for everything below this in the tree
+        return ProviderScope(
+          overrides: [
+            todoItemIndexProvider.overrideWithValue(index),
+          ],
+          // Using this approach, TodoWidget can now be constant
+          // since no index needs to be passed down
+          child: const TodoWidget(),
+        );
+      },
+    );
+  }
+}
+```
+
+Next we will create a family provider to provide the actual todo object based on their index. 
+This is unnecessary since we could just access the `todoListProvider` directly, 
+but it will be useful to have a separate provider for our refactor.
+```dart
+/// Provides a single todo, so you can watch them individually
+///
+/// This way you will only be notified if the particular todo you
+/// were watching was updated
+final todoProvider = Provider.family<Todo, int>(
+    (ref, index) => ref.watch(todoListProvider).state[index]);
+```
+
+Finally we will create a widget to display the todo.
+Note that it watches the index that has been overridden for the current subtree
+and uses that index to access the particular todo in our family provider.
+
+Already scoping has made it so that our `TodoWidget` can be `const` rather than having to pass down the index.
+While this is premature optimization, it is a useful technique to be aware of. 
+The `TodoWidget` uses it's context and location in the tree to determine which `Todo` to show, rather than using a parameter.
+```dart
+
+/// A widget to show a single todo
+class TodoWidget extends ConsumerWidget {
+  const TodoWidget();
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    // Gets the index of this particular Todo
+    final index = ref.watch(todoItemIndexProvider);
+    print('Building Item $index');
+    // Watches a single todo item
+    final todoItem = ref.watch(todoProvider(index));
+    // Display it!
+    return Text(todoItem.todo);
+  }
+}
+```
+Finally let's create a main function that shows the list and has a `FloatingActionButton` that adds a new todo item to the list.
+```dart
+void main() {
+  runApp(
+    ProviderScope(
+      child: MaterialApp(
+        home: Scaffold(
+          body: TodoListWidget(),
+          floatingActionButton: Consumer(builder: (context, ref, child) {
+            return FloatingActionButton(
+              onPressed: () {
+                ref
+                    .read(todoListProvider)
+                    .update((state) => [...state, Todo('New Todo')]);
+              },
+              child: Icon(Icons.add),
+            );
+          }),
+        ),
+      ),
+    ),
+  );
+}
+```
+
+Running it should print
+
+```
+flutter: Build List
+flutter: Build Item 0
+flutter: Build Item 1
+flutter: Build Item 2
+// After updating item 0 in the todoListProvider by pressing the FloatingActionButton
+// you will then see:
+flutter: Build Item 3
+```
+As can be seen in the logs, flutter does not need to build every item in the list.
+
+
+## Scoping Dependent Providers
```

Comment:
A valuable feature to cover would be `ProviderScope.parent`, combined with showing modals.